### DISCUSSION TITLE
Better tools for electric rails / electrification

### DISCUSSION THREAD

**Discussion by victorleblais** [voxeltycoon/issues#754] [UI&UX]
**Is your feature request related to a problem? Please describe.**
It can be sometimes pretty hard to electrify pieces of rails, especially inside intersections.
Example by Alban_fooz on Discord
![image](https://user-images.githubusercontent.com/4365075/118288144-67a70e80-b4d4-11eb-8879-a46c2c5e88c0.png)

The current tool changes the rail state between three modes:
- No electrification
- Electrified with poles on right side
- Electrified with poles on left side

**Describe the solution you'd like**
The Electrification tool could be reworked into 2 different tools:
- Electrify: Toggles electrification on/off
- Electric poles side: Changes the side of the electric poles

The tools would greatly benefit from having a custom "white-mode" view similar to the view when placing signals.
- Non-electrified tracks could be highlighted in red
- When hovering a segment of track, it could be highlighted in yellow, pressing Ctrl or Shift would work too
- Electrified segment of track could be highlighted in green

Additionally, instead of clicking on tracks, it would be great to be able to draw a rectangle (similar to how the terraforming works) by dragging the mouse cursor starting from an empty square (not a piece of track). The rectangle would:
- highlight all the tracks that it would electrify
- display the cost in a tooltip, updated in real time according to the size/shape of the rectangle

Once the cursor is released, all the tracks inside the rectangle are electrified.
Maybe pressing CTRL when drawing the rectangle could remove electrification.

**Comment by albanfooz** [comment]
I was thinking of other ways to facilitate electrifying tracks : 
- Electrify block delimited by signals
  For example, it would electrify every rail section of the selected block (this way of electrifying tracks would be the most effective on the yellow block)
![20210514203241_1](https://user-images.githubusercontent.com/33288174/118313874-ac8e6d80-b4f3-11eb-8441-843d6c808db2.jpg)
- Electrify via the "Lay rails" tool : if you lay a train track with _Electrification_ \*ON\*, on already existing tracks it electrifies it. Like this : 
![20210514202925_1](https://user-images.githubusercontent.com/33288174/118313571-325de900-b4f3-11eb-8e79-014aa5ae3593.jpg)

**Comment by gralewski** [comment]
Simpler solution would be - electrify whole connected tracks - not track segments. Maybe using modifier like shift etc. The game already knows what is connected to what. Not having any overlay, just these poles makes electrification hard.

**Comment by gralewski** [comment]
Also - since electrified tracks are already downwards compatible - there could be an option to lay already electrified tracks just as we invent it.
